Added Cross System Support for Portmap

thebuilder6 opened the pull request:
@UxuginPython This was an attempt to clean up your port map changes, but I did not really understand your vision for in the Devices class. So If you want to look at that again you could figure out what you want keep from this and yours

thebuilder6 committed:
Added Cross System Support for Portmap

@UxuginPython This was an attempt to clean up your port map changes, but I did not really understand what you we going for in the Devices class

Co-Authored-By: UxuginPython <84234447+UxuginPython@users.noreply.github.com>

UxuginPython commented:
@thebuilder6 Thank you for cleaning this up! I'm sorry I haven't been online; we had some internet trouble. The main point of the Device class is to improve compatibility between robots and scripts. For example, if a script just needs a robot with a drive base, without Device you would need to check with something like `hubName in ['move', 'primesnow', 'primebug']`, but with Device you could just do `device.supports_drive_base`. This is a bit easier to understand and only PortMap needs to be updated for a new design.

thebuilder6 committed:
Added Device Class and Split in to two folders

I Added a Device Class that accomplishes similar to what you had(I think its pretty clever) and Split it in to two folders so that the simple stuff is at the top of PortMap

Co-Authored-By: UxuginPython <84234447+UxuginPython@users.noreply.github.com>

thebuilder6 commented:
@UxuginPython I Added a Device Class that accomplishes similar to what you had(I think its pretty clever), but not necessarily all of it so it might need to be changed later. And I split it in to two folders so that the simple stuff is at the top of PortMap to make it easier for the less experienced coders. I was not sure what you were doing with the __str__ capabillity, so I just replicated it in the Device class. Let me know if we should merge to main or if you want to work on it some more

thebuilder6 committed:
Fixed hub issue

made hub global and passed to PortMap

UxuginPython committed:
Add files via upload

see #8

thebuilder6 committed:
Merge pull request #9 from Snowbotics39131/PortMap-EV3-support

EV3 support, and a few other things
